packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -597,13 +602,18 @@ export function validateBlobsAndTrees(snapshot: IOdspSnapshot): void {
 	);
 }
 
-function countTreesInSnapshotTree(snapshotTree: ISnapshotTree): number {
-	let numTrees = 0;
-	for (const [_, tree] of Object.entries(snapshotTree.trees)) {
-		numTrees += 1;
-		numTrees += countTreesInSnapshotTree(tree);
+function getTreeStatsCore(snapshotTree: ISnapshotTree, stats: ITreeStats): void {
+	stats.blobNodes += Object.entries(snapshotTree.blobs).length;
+
+	const entries = Object.entries(snapshotTree.trees);
+	if (entries.length === 0) {
+		stats.leafTrees++;
```

Comment:
Should we also do stats.trees++ here too as trees should be total trees and leafTrees are just leaf trees?

Reply:
It's less about entries.length === 0 case, it's more a question - should we do it unconditionally.
The old code did not do it, i.e. it counted only non-leaf trees.
I think it makes sense to do so, even though it changes semantics of that property -  will do so